### PR TITLE
13.0.0

### PR DESCRIPTION
**Pull request by kay0u**
## Solution
- *[Reduced memory consumption of GitLab with Puma](https://about.gitlab.com/releases/2020/05/22/gitlab-13-0-released/#reduced-memory-consumption-of-gitlab-with-puma)*
- *[13.0.0](https://about.gitlab.com/releases/2020/05/22/gitlab-13-0-released/)*

## PR Status
- [x] Code finished.
- [x] Tested with Package_check.
- [x] Fix or enhancement tested.
- [x] Upgrade from last version tested.
- [x] Can be reviewed and tested.

## Validation
---
- [x] **Code review** : yalh76
- [x] **Approval (LGTM)** :  yalh76
*Code review and approval have to be from a member of @YunoHost-Apps/apps-group*
- **CI succeeded** : 
[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/gitlab_ynh%20PR114%20(Kay0u)/) 
When the PR is marked as ready to merge, you have to wait for 3 days before really merging it.

### CODE REVIEW

File: scripts/upgrade [hunk 1]
```diff
@@ -242,26 +254,47 @@ then
 	source ./upgrade.d/upgrade.last.sh
 	last_version=$gitlab_version
 
+	source_current_major_version () {
+		if [ -e "./upgrade.d/upgrade.$current_major_version.first.sh" ]; then
+			source ./upgrade.d/upgrade.$current_major_version.first.sh
+		elif [ -e "./upgrade.d/upgrade.$current_major_version.last.sh" ]; then
+			source ./upgrade.d/upgrade.$current_major_version.last.sh
+		fi
```
Comment (kay0u) [inline scripts/upgrade:262]:
```suggestion
		else
			source ./upgrade.d/upgrade.last.sh
		fi
```

File: scripts/upgrade [hunk 2]
```diff
@@ -242,26 +254,47 @@ then
...
+				current_major_version=$(($current_major_version + 1))
+				source_current_major_version
+			fi
 		fi
 
 		# Finish with the last migration if the file doesn't exist
-		if [ ! -e "./upgrade.d/upgrade.$current_major_version.sh" ]; then
-			current_major_version=last
+		if [ ! -e "./upgrade.d/upgrade.$current_major_version.first.sh" ] && [ ! -e "./upgrade.d/upgrade.$current_major_version.last.sh" ]; then
+			source ./upgrade.d/upgrade.last.sh
```
Comment (kay0u) [inline scripts/upgrade:295]: we can remove this with my suggestion (I have to test it)

**Comment by kay0u** [comment]
Merging in 3 days

**Comment by Christophe31** [comment]
maybe late but a critical security patch issued (13.0.4) https://about.gitlab.com/releases/2020/06/03/critical-security-release-13-0-4-released/
May be relevant.

**Comment by kay0u** [comment]
Let's merge that PR, I'll create a new one right after.

**Pull request merged**